### PR TITLE
Rule 1.2: Address false positives for variable length arrays, improve reporting

### PR DESCRIPTION
**Pull request by lcartey**
## Description

 - Improves reporting by providing a more descriptive message of which C extensions were used.
 - Improves reporting by grouping results generated from a macro and reporting them at the generating macro location instead.
 - Fixes https://github.com/github/codeql-coding-standards/issues/701.

## Change request type

 - [ ] Release or process automation (GitHub workflows, internal scripts)
 - [ ] Internal documentation
 - [ ] External documentation
 - [x] Query files (`.ql`, `.qll`, `.qls` or unit tests)
 - [ ] External scripts (analysis report or other code shipped as part of a release)

## Rules with added or modified queries

 - [ ] No rules added
 - [ ] Queries have been added for the following rules:
 - [x] Queries have been modified for the following rules:
     - `RULE-1-2`

## Release change checklist

A change note ([development_handbook.md#change-notes](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#change-notes)) is required for any pull request which modifies:

 - The structure or layout of the release artifacts.
 - The evaluation performance (memory, execution time) of an existing query.
 - The results of an existing query in any circumstance.

If you are only adding new rule queries, a change note is not required.

_**Author:**_ Is a change note required? 
  - [x] Yes
  - [ ] No

🚨🚨🚨
_**Reviewer:**_ Confirm that format of *shared* queries (not the .qll file, the
.ql file that imports it) is valid by running them within VS Code. 
  - [x] Confirmed

_**Reviewer:**_ Confirm that either a change note is not required or the change note is required and has been added.
  - [x] Confirmed

## Query development review checklist

For PRs that add new queries or modify existing queries, the following checklist should be completed by both the author and reviewer:

### Author

 - [x] Have all the relevant rule package description files been checked in?
 - [x] Have you verified that the metadata properties of each new query is set appropriately?
 - [x] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [x] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [x] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [x] Does the query have an appropriate level of in-query comments/documentation?
 - [x] Have you considered/identified possible edge cases?
 - [x] Does the query not reinvent features in the standard library?
 - [x] Can the query be simplified further (not golfed!)

### Reviewer 

 - [x] Have all the relevant rule package description files been checked in?
 - [x] Have you verified that the metadata properties of each new query is set appropriately?
 - [x] Do all the unit tests contain both "COMPLIANT" and "NON_COMPLIANT" cases?
 - [x] Are the alert messages properly formatted and consistent with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#query-style-guide)?
 - [x] Have you run the queries on OpenPilot and verified that the performance and results are acceptable?<br />_As a rule of thumb, predicates specific to the query should take no more than 1 minute, and for simple queries be under 10 seconds. If this is not the case, this should be highlighted and agreed in the code review process._
 - [x] Does the query have an appropriate level of in-query comments/documentation?
 - [x] Have you considered/identified possible edge cases?
 - [x] Does the query not reinvent features in the standard library?
 - [x] Can the query be simplified further (not golfed!)

### CODE REVIEW

File: c/misra/test/rules/RULE-1-2/LanguageExtensionsShouldNotBeUsed.expected [hunk 1]
```diff
@@ -1,51 +1,51 @@
...
+| test.c:113:11:113:16 | ... ? ... : ... | Ternaries with omitted middle operands are a compiler extension and is not portable to other compilers. |
+| test.c:124:12:124:12 | definition of a | 128-bit integers are a compiler extension and are not portable to other compilers. |
+| test.c:128:17:128:17 | definition of a | Double-Word integers are a compiler extension and are not portable to other compilers. |
+| test.c:165:8:165:15 | definition of contents | Zero length arrays are a compiler extension and are not portable to other compilers. |
+| test.c:182:8:182:11 | gf19 | Empty structures are a compiler extension and are not portable to other compilers. |
+| test.c:216:9:216:10 | definition of x1 | Zero length arrays are a compiler extension and are not portable to other compilers. |
+| test.c:266:16:266:21 | access | Use of attribute  'access' is a compiler extension and is not portable to other compilers. |
+| test.c:270:5:270:9 | alias | Use of attribute  'alias' is a compiler extension and is not portable to other compilers. |
```
Comment (nicolaswill) [inline c/misra/test/rules/RULE-1-2/LanguageExtensionsShouldNotBeUsed.expected:26]:
```suggestion
| test.c:269:27:269:31 | alias | Use of attribute  'alias' is a compiler extension and is not portable to other compilers. |
```

**Review by nicolaswill — CHANGES_REQUESTED** [review]
LGTM except for the mismatched location on one of the expected test results.

**Review by nicolaswill — APPROVED** [review]
